components/micro-gateway-core/src/main/ballerina/gateway/basicAuth.bal

```diff
@@ -0,0 +1,145 @@
+// Copyright (c)  WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file   except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerina/auth;
+import ballerina/config;
+import ballerina/runtime;
+import ballerina/system;
+import ballerina/time;
+import ballerina/io;
+import ballerina/reflect;
+import ballerina/crypto;
+
+@Description { value: "Representation of the BasicAuth logic" }
+public type BasicAuthUtils object {
+
+    public http:AuthnHandlerChain authnHandlerChain;
+    public new(authnHandlerChain) {}
+
+    @Description { value: "representation of Basic auth validation" }
+    @Param { value: "listener: Listner endpoint" }
+    @Param { value: "request: Request instance" }
+    @Param { value: "context: FilterContext instance" }
+    @Return { value: "FilterResult:BasicAuth result to indicate which folw is selected for request to proceed" }
+    public function processRequest(http:Listener listener, http:Request request, http:FilterContext context)
+                        returns boolean {
+
+        boolean isAuthenticated;
+        //API authentication info
+        AuthenticationContext authenticationContext;
+        boolean isAuthorized;
+        string[] providerIds = [AUTHN_SCHEME_BASIC];
+        //set Username from the request
+        string authHead = request.getHeader(AUTHORIZATION_HEADER);
+        string[] headers = authHead.trim().split("\\s* \\s*");
+        string encodedCredentials = headers[1];
+        var decodedCredentials = encodedCredentials.base64Decode();
+        //Extract username and password from the request
+        string userName;
+        string passWord;
+        match decodedCredentials {
+            string decoded => {
+                if(!decoded.contains(":")){
+                    setErrorMessageToFilterContext(context, API_AUTH_BASICAUTH_INVALID_FORMAT);
+                    sendErrorResponse(listener, request, untaint context);
+                    return false;
+                }
+                string[] decodedCred = decoded.trim().split(":");
+                userName = decodedCred[0];
+                if(lengthof decodedCred < 2){
```

Comment:
Formatting issue. Lets format the code. Lets check for formatting issues in other places as well.